File: placementRush/11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& nums) {
        int n = nums.size();
        int left = 0; 
        int right = n-1;
        int maxi = -1e9;
        while(left<right){
            int smaller = min(nums[left], nums[right]);
            int range = right - left;
            maxi = max(maxi, smaller*range);
            if(smaller == nums[left]){
                left++;
            }else{
                right--;
            }
        }
        return maxi;
    }
};

File: placementRush/11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        maxi = 0
        j = len(height) -1
        i = 0
        while(i<j):
            area = min(height[i], height[j]) * (j - i)
            if area > maxi:
                maxi = area
            if (height[i]<height[j]):
                i = i+1
            else:
                j = j-1
        return maxi